src/main/java/org/wso2/carbon/esb/connector/NatsConnection.java

```diff
@@ -0,0 +1,253 @@
+        String traceConnection = (String) messageContext.getProperty(NatsConstants.TRACE_CONNECTION);
+        String useOldRequestStyle = (String) messageContext.getProperty(NatsConstants.USE_OLD_REQUEST_STYLE);
+        String noRandomize = (String) messageContext.getProperty(NatsConstants.NO_RANDOMIZE);
+        String noEcho = (String) messageContext.getProperty(NatsConstants.NO_ECHO);
+
+        Properties serverConfig = new Properties();
+
+        // Default values for these properties are empty strings
+        serverConfig.setProperty(Options.PROP_SERVERS, servers);
+        serverConfig.setProperty(Options.PROP_USERNAME, username);
+        serverConfig.setProperty(Options.PROP_PASSWORD, password);
+        serverConfig.setProperty(Options.PROP_CONNECTION_NAME, connectionName);
+        serverConfig.setProperty(Options.PROP_VERBOSE, verbose);
+        serverConfig.setProperty(Options.PROP_PEDANTIC, pedantic);
+        serverConfig.setProperty(Options.PROP_UTF8_SUBJECTS, supportUtf8Subjects);
+        serverConfig.setProperty(Options.PROP_USE_OLD_REQUEST_STYLE, useOldRequestStyle);
+        serverConfig.setProperty(Options.PROP_NORANDOMIZE, noRandomize);
+        serverConfig.setProperty(Options.PROP_NO_ECHO, noEcho);
+
+        // Setting these properties with null will throw NullPointerException or setting them to an empty string will cause errors
+        setServerConfigProperty(serverConfig, Options.PROP_INBOX_PREFIX, inboxPrefix);
+        setServerConfigProperty(serverConfig, Options.PROP_RECONNECT_BUF_SIZE, reconnectBufferSize);
+        setServerConfigProperty(serverConfig, Options.PROP_RECONNECT_WAIT, reconnectWait);
+        setServerConfigProperty(serverConfig, Options.PROP_MAX_RECONNECT, maxReconnects);
+        setServerConfigProperty(serverConfig, Options.PROP_CONNECTION_TIMEOUT, connectionTimeout);
+        setServerConfigProperty(serverConfig, Options.PROP_MAX_CONTROL_LINE, maxControlLine);
+        setServerConfigProperty(serverConfig, Options.PROP_PING_INTERVAL, pingInterval);
+        setServerConfigProperty(serverConfig, Options.PROP_CLEANUP_INTERVAL, requestCleanUpInterval);
+        setServerConfigProperty(serverConfig, Options.PROP_MAX_PINGS, maxPingsOut);
+        setServerConfigProperty(serverConfig, Options.PROP_DATA_PORT_TYPE, dataPortType);
+
+        Options.Builder builder = new Options.Builder(serverConfig);
+
+        if (StringUtils.isNotEmpty(bufferSize)) {
+            builder.bufferSize(Integer.parseInt(bufferSize));
+        }
+
+        if (Boolean.parseBoolean(turnOnAdvancedStats)) {
+            builder.turnOnAdvancedStats();
+        }
+
+        if (Boolean.parseBoolean(traceConnection)) {
+            builder.traceConnection();
+
+        }
+
+        if (StringUtils.isNotEmpty(tlsProtocol + tlsTrustStoreType + tlsTrustStoreLocation + tlsTrustStorePassword + tlsKeyStoreType + tlsKeyStoreLocation + tlsKeyStorePassword + tlsKeyManagerAlgorithm + tlsTrustManagerAlgorithm)) {
```

Comment:
Better to use string builder

Reply:
The compiler will automatically use StringBuilder right?